### ISSUE TITLE
There is a problem about BN layer.

### ISSUE THREAD

**Issue by huang229** [CoinCheung/BiSeNet#37]
Thank you very much for your open source sharing.
Why do you write the BN layer script alone? Isn't there a defined BN in Pytorch? Is there something wrong with the BN layer of Pytorch? I use the BN layer of pytorch. When the size of batch_size is too small, the model test is very bad. I have never encountered this problem in other deep learning frameworks.

**Comment by CoinCheung** [comment]
I used the inplace-abn which helps to save memory, you can also replace it with the official bn implementation. 

BN has the problem of performance decay when batchsize is small, one method to cope with it is to use sync-bn which normalize the activations on the whole batch rather than on each gpu.

**Comment by huang229** [comment]
I replaced the sync BN layer with pytorch's BN layer. Batch_size = 8, crossize = [640, 480], the loss value only dropped to 3.8172. as follows：
Do I have any training skills that I didn't notice?
[epoch 215][step 79800] , lr: 4.62e-05, time cost 96.5s
loss_avg: 3.8186
[epoch 215][step 79850] , lr: 3.57e-05, time cost 94.8sloss_avg: 3.8182
[epoch 215][step 79900] , lr: 2.49e-05, time cost 94.1sloss_avg: 3.8179
[epoch 215][step 79950] , lr: 1.35e-05, time cost 94.4s  loss_avg: 3.8176
[epoch 215][step 80000] , lr: 3.91e-07, time cost 94.6s  loss_avg: 3.8172

**Comment by huang229** [comment]
@CoinCheung

**Comment by CoinCheung** [comment]
what is the final mIOU, does it looks reasonable?

**Comment by huang229** [comment]
@CoinCheung I'm really sorry. I've been busy. I'll wait until today to reply. 
Because my GPU capacity is very small, I have changed some parameters:
batch_size =8
cropsize = [640, 480]
Replace the batch norm with the official interface provided by pytorch.
After several days of repeated training, test model epoch = 212
Minimum loss value = 3.09
I verified the accuracy of the test. mIOU =  0.6836137883275449

Although I have reduced the input image size of the training, I feel that the accuracy is still very low.
In addition, when the loss value drops to about 3, it will not change any more. I think it seems that there are some problems. I would like to ask how much your loss value has decreased. Can I tell you conveniently?

**Comment by huang229** [comment]
In addition, I also think that the improvement of accuracy is not only related to the input size of the test data, but also to the input size of the training data.I can only see that the original author's paper mentioned the different precision brought by changing the input data size during testing, but did not mention the input data size during training. Do you see it?

**Comment by CoinCheung** [comment]
Yes, you are right. The training crop size matters. You can refer to the paper of [efficientnet](https://arxiv.org/abs/1905.11946) for the scaling relationship between performance and input image sizes.

**Comment by CoinCheung** [comment]
Closing this since you might have figured out the solutions.
